routes/xref/update.js

```diff
@@ -0,0 +1,81 @@
+const bodyParser = require("body-parser");
+const crypto = require("crypto");
+const { exec } = require("child_process");
+
+const bikeshedSecret = process.env.BIKESHED_SECRET;
+if (!bikeshedSecret) {
+  throw new ReferenceError("env variable `BIKESHED_SECRET` is not set.")
+}
+
+// minimum delay between subsequent updates
+const UPDATE_THRESHOLD = 60 * 60 * 1000; // 1 hour

```

Comment:
I'm not a fan of this, bth. Seems very arbitrary. What we are trying to prevent is multiple executions of `exec`, but not setting some timeout limit. It might be the case the BS data pushes a buggy update, and then a few seconds later they push another update that fixes the data.